geoh5py/shared/concatenation/object.py

```diff
@@ -38,14 +36,39 @@ class ConcatenatedObject(Concatenated, ObjectBase, ABC):
     _property_groups: list | None = None
 
     def __init__(self, entity_type, **kwargs):
-        if kwargs.get("parent") is None or not is_concatenator(kwargs.get("parent")):
+        if kwargs.get("parent") is None:
             raise UserWarning(
                 "Creating a concatenated object must have a parent "
                 "of type Concatenator."
             )
 
         super().__init__(entity_type, **kwargs)
 
+    def add_children(
+        self,
+        children,
+    ):
+        """
+        :param children: Add a list of entities as
+            :obj:`~geoh5py.shared.entity.Entity.children`
+        """
+        property_groups = self._property_groups or []
+
+        for child in children:
+            if child not in self._children and isinstance(
+                child, (ConcatenatedData, ConcatenatedPropertyGroup)
+            ):
+                self._children.append(child)
+
+            if (
+                isinstance(child, ConcatenatedPropertyGroup)
+                and child not in property_groups
+            ):
+                property_groups.append(child)
+
+            if property_groups:
```

Comment:
But then if the type is not the one we want, it just pass, not raising an error or anything?

Reply:
Currently, it won't get appended to the list. We never raised the error before in the parent class `ObjectBase`, but we could.

Reply:
The add_data of entity accepts everything (which is strange for a public method but out of the scope of this project). 

the one of object passes, I agree, but a warning could be nice. :-)

Reply:
I agree, Entity probably shouldn't have the add_children, since Data is also a sub-class but can't have children (as far as I know). Feel free to raise a specific issue for this.